dev-docs/RFCs/v4.0.0/terms_for_concepts_and_datatypes.md

```diff
@@ -0,0 +1,96 @@
+
+See, for example: https://github.com/uber/h3-js/issues/53
+
+
+## Names for H3 Edge Types
+
+Instead of `UnidirectionalEdge`, use the term `DirectedEdge`.
+
+For a future undirected edge mode, use the term `Edge`.
+
+Function name changes:
+
+|                  Current name                 |        Proposed name         |
+|-----------------------------------------------|------------------------------|
+| `getH3UnidirectionalEdge`                     | `getDirectedEdge`            |
+| `h3UnidirectionalEdgeIsValid`                 | `isValidDirectedEdge`        |
+| `getOriginH3IndexFromUnidirectionalEdge`      | `getDirectedEdgeOrigin`      |
+| `getDestinationH3IndexFromUnidirectionalEdge` | `getDirectedEdgeDestination` |
+| `getH3IndexesFromUnidirectionalEdge`          | `getDirectedEdgeCells`       |
+| `getH3UnidirectionalEdgesFromHexagon`         | `getDirectedEdgesFromCell`   |
+| `getH3UnidirectionalEdgeBoundary`             | `getDirectedEdgeBoundary`    |
+
+
+## Function Names
+
+A few more function name change proposals:
+
+|          Current name         |     Proposed name     |                  Notes                  |
+|-------------------------------|-----------------------|-----------------------------------------|
+| (doesn't exist)               | `isValidIndex`        |                                         |
+| `h3IsValid`                   | `isValidCell`         |                                         |
+| `h3UnidirectionalEdgeIsValid` | `isValidDirectedEdge` |                                         |
+| `h3IsPentagon`                | `isPentagon`          |                                         |
+| `h3ToParent`                  | `cellToParent`        |                                         |
+| `h3ToChildren`                | `cellToChildren`      |                                         |
+| `h3IndexesAreNeighbors`       | `cellsAreNeighbors`   |                                         |
+| `geoToH3`                     | `geoToCell`           |                                         |
+| `h3ToGeo`                     | `cellToGeo`           |                                         |
+| `kRing`                       | `disk`                | filled-in disk                          |
+| `hexRing`                     | `ring`                | hollow ring                             |
+| `numHexagons`                 | `numCells`            |                                         |
+| `getRes0Indexes`              | `getRes0Cells`        |                                         |
+| `getPentagonIndexes`          | `getPentagons`        |                                         |
+| `h3GetBaseCell`               | `getBaseCellNumber`   |                                         |
+| `h3Distance`                  | `gridDistance`        | no `get` prefix?                        |
+| `h3Line`                      | `cellPath`            | future: `edgePath`, `directedEdgePath`? |
```

Comment:
Agreed.

How about `gridDistance` and `gridPath`?

**Or**, should it be `gridPathCell`, anticipating possible future functions like `gridPathEdge` and `gridPathDirectedEdge`?

Reply:
I think `grid` here implies `cell` (if not, we should use `cellDistance` and `cellPath`). If we ever wanted to provide an edge path, I'd go with e.g. `directedEdgePath`.

Reply:
I was thinking that `grid` implies the cell/edge **graph**, but not specifically the `cell` (the nodes in the graph).

`cellDistance` and `cellPath` don't make me think of the graph, necessarily. 

The same "grid path" between two cells could be represented a number of ways:

- an ordered sequence of cells (`gridPath` or `gridPathCell`)
- a collection of directed edges (`gridPathDirectedEdge`)
- a collection of undirected edges (`gridPathEdge`)

Reply:
I guess I don't see where the addition of the term `grid` for these functions adds any clarity beyond what's implied by `cellPath`, `edgePath`, etc. Cells, edges, and directed edges are already in a grid by definition.

Reply:
Above, I thought you were concerned about "I feel like `h3Line` and `h3Distance` are closely related, but this naming doesn't reflect that.", so I was trying to find a naming that made that connection clear, but also allowed for the different potential path representations.

`cellPath` doesn't evoke the grid in my mind. As the fundamental unit of DGGS, that feels like it stands on its own. That is, `grid` implies `cell`, `edge`, and `path`, but the converse doesn't seem natural to me.

Specifically, `cellDistance` could be interpreted as the Haversine distance between cell centroids, but `gridDistance` eliminates that ambiguity for me.

I kind of like `gridDistance`, `cellPath`, `edgePath`, and `directedEdgePath`, but that loses the connection between distance and path, as you noted above.

Reply:
I also like having related functions change faster in the right-most characters, which is what drew me to `___PathCell`, `___PathDirectedEdge`, `___PathEdge`. But that's more of a personal preference :)

Reply:
Added some new ideas!

Reply:
While I've always used `gridDistance` as you do here, maybe it would make sense to tie the distance directly to the rest of the terminology by using `cellPathDistance`.

Reply:
I'm now proposing:

| Current name |     Proposed name      |
|--------------|------------------------|
| `h3Distance` | `gridDistance`         |
| `h3Line`     | `gridPathCell`         |
| *DNE*        | `gridPathEdge`         |
| *DNE*        | `gridPathDirectedEdge` |

I think I'm happy with `gridDistance` instead of `cellPathDistance`, as the concept of "distance" is the same for the following three representations of the same path.